packages/components/pie-modal/src/index.ts

```diff
@@ -57,60 +132,59 @@ export class PieModal extends RtlMixin(LitElement) {
     }
 
     /**
-     * We require this because toggling the prop `isOpen` itself won't
-     * allow the dialog to open in the correct way (with the default background),
-     * the method `showModal()` needs to be invoked.
-     *
-     * https://lit.dev/docs/components/lifecycle/#willupdate
+     * Dismisses the modal on backdrop click
      *
-     * @param changedProperties
      */
-    willUpdate (changedProperties: Map<string, unknown>) {
-        if (changedProperties.has('isOpen') && this._dialog) {
-            const dialog = this._dialog;
-            const isClosed = changedProperties.get('isOpen') === false;
+    private _handleDialogLightDismiss = (event: MouseEvent) : void => {
+        const rect = this._dialog?.getBoundingClientRect();
 
-            if (isClosed) {
-                dialog.showModal();
-            } else {
-                dialog.close();
-            }
+        const {
+            top = 0, bottom = 0, left = 0, right = 0,
+        } = rect || {};
+
+        // This means the dialog is not open due to clicking the close button so
+        // we want to escape early
+        if (top === 0 && bottom === 0 && left === 0 && right === 0) {
+            return;
         }
-    }
 
-    _handleCloseDialog = () => {
-        this._dialog?.close();
-        this._onDialogClose();
+        const isClickOutsideDialog = event.clientY < top ||
+            event.clientY > bottom ||
+            event.clientX < left ||
+            event.clientX > right;
+
+        if (isClickOutsideDialog) {
+            this.isOpen = false;
+        }
     };
 
     /**
-     * Dismiss modal via `_handleCloseDialog()` on backdrop click
-     * and will proceed to fire an `ON_MODAL_CLOSE_EVENT` event.
+     * Dispatch `ON_MODAL_CLOSE_EVENT` event.
+     * To be used whenever we close the modal.
      *
+     * @event
      */
-    private _handleDialogLightDismiss = (event: MouseEvent) => {
-        const rect = this._dialog?.getBoundingClientRect();
-
-        if (typeof rect !== 'undefined') {
-            const isClickOutsideDialog = event.clientY < rect.top ||
-                event.clientY > rect.bottom ||
-                event.clientX < rect.left ||
-                event.clientX > rect.right;
+    private _dispatchModalCloseEvent = () : void => {
```

Comment:
Super late to this game (sorry, I missed you were looking for re-reviews).

Would be good at some point to have a `dispatchEvent` helper, to save with the repetition when dispatching events.

Not sure if we should just do this as part of the core helpers potentially as it'll be a cross-component thing I imagine?

Reply:
yep fully agree with this

Reply:
We should be able to completely get rid of some of these dispatching event calls, im planning on cleaning some up in my PR. With regards to sharing I think we should move to a helper file which will be nice like you said.